---
title: "Projects"
layout: single
permalink: /projects/
author_profile: true
classes: wide smaller-font
---

### Benchmarking Protein–Ligand Binding Site Prediction with Pseq2Sites
Bachelor thesis project benchmarking deep learning approaches (CNN with attention mechanisms) for sequence-based prediction
 of protein–ligand binding sites. Focused on large-scale preprocessing of scPDB and PDBbind datasets, implementation of 
ProtTrans embeddings, and comparative evaluation to highlight strengths, challenges, and opportunities for improving drug 
discovery pipelines. [GitHub Link](https://github.com/AsalRb/Benchmarking_Protein-Ligand_Binding_Site_Prediction_with_Pseq2Sites)


### Exploring Relationships Between Ion Channels and lncRNAs in Gastric Cancer
Ongoing research project using TCGA-STAD RNA-seq data to investigate gene co-expression networks. Applied differential 
expression analysis (DESeq2), WGCNA, and survival modeling to identify lncRNA–ion channel modules correlated with clinical
 traits, aiming to uncover novel biomarkers and therapeutic targets. [GitHub Link](https://github.com/AsalRb/Exploring_Relationships_Between_Ion_Channels_and_lncRNAs_in_Gastric_Cancer)


### Genetic Database
Designed and implemented a normalized relational database (up to BCNF) for efficient genetic data management. Built a 
complete pipeline including entity–relationship modeling, schema design, and SQL implementation in PostgreSQL 
(via pgAdmin4). The system supports storing and querying of genes, RNAs, proteins, organisms, and users, with 
role-based access control and example datasets.[GitHub Link](https://github.com/AsalRb/Genetic_Database)


### Read Mapping and Genome Assembly
Implemented a full NGS workflow on \textit{E. coli} short-read data, including quality control (FastQC), de novo assembly
 (SPAdes, Quast), and read mapping (BWA, SAMtools). Validated results by visualizing alignments in IGV and assessing 
concordance, mapping rates, and read depth, providing hands-on experience with genome assembly and evaluation. [GitHub Link](https://github.com/AsalRb/Read_Mapping_and_Genome_Assembly)


### Identification of Xylanase Genes
Designed and implemented a bioinformatics pipeline to identify thermostable xylanase genes from rumen metagenomic 
datasets. The workflow included sequence similarity searches (BLAST+), clustering at 97% identity with CD-HIT to 
generate non-redundant representatives, multiple sequence alignment (Clustal Omega) to detect conserved motifs, and 
HMM-based filtering to refine candidates. Produced a curated set of high-confidence xylanase gene sequences. [GitHub Link](https://github.com/AsalRb/Identification_of_Xylanase_Genes_from_Rumen_Metagenome)


### Hierarchical Data Structure for Genetic Diseases
Developed a Python-based tree system to organize genetic diseases and associated genes. Implemented search and 
insertion functions with hash tables for gene attributes (location, exons, type, name) and a menu-driven interface,
 enabling efficient hierarchical representation of biomedical data. [GitHub Link](https://github.com/AsalRb/Hierarchical_Data_Structure_for_Genetic_Diseases_Using_Tree_Algorithms)


